app/app/tests/api/api_v1/test_courses.py

```diff
@@ -0,0 +1,145 @@
+from typing import Dict
+
+from sqlalchemy.orm import Session
+from starlette.testclient import TestClient
+
+from app import crud
+from app.core.config import settings
+from app.schemas.users.admin import AdminPermissions
+from app.tests.utils.course import create_random_course, create_random_term
+from app.tests.utils.user import authentication_token_from_email, create_random_user
+from app.tests.utils.utils import random_lower_string, to_json
+
+
+def test_get_all_courses(client: TestClient, superuser_token_headers: Dict[str, str], db: Session) -> None:
+    course = create_random_course(db)
+    r = client.get(f"{settings.API_V1_STR}/courses/", headers=superuser_token_headers)
+    assert r.status_code == 200
+    results = r.json()
+    assert results
+    assert results[-1]['name'] == course.name
+    assert results[-1]['code'] == course.code
+    assert results[-1]['term_id'] == course.term_id
+
+
+def test_get_all_courses_with_details(client: TestClient, superuser_token_headers: Dict[str, str], db: Session) -> None:
```

Comment:
what is this? I thought `details` does nothing rn? Since we're returning the term details by default anyway

Reply:
Need to remove the test before this test. As you said, I followed what you guys had done the previous PR. The relation one. That had this separate. 

Check years api tests in master. I followed what was changed in previous PR.

Reply:
Oh wow, I didn't realise those were leftover tests